.github/workflows/ci.yml

```diff
@@ -53,9 +53,9 @@ jobs:
       - name: Unit and e2e tests
         run: pnpm test
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         if: always()
         with:
-          name: playwright-report
-          path: playwright-report/
+          name: playwright-report-${{ matrix.runs-on }}
+          path: playwright-report-${{ matrix.runs-on }}/
```

Comment:
Following the suggestion in the migration guide here: https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md